components/nexus-common/src/main/java/org/sonatype/nexus/common/entity/Entity.java

```diff
@@ -24,6 +24,11 @@
   private EntityMetadata metadata;
 
   @Nullable
+  public EntityId id() {
+    return getEntityMetadata() != null ? getEntityMetadata().getId() : null;
+  }
```

Comment:
I was trying to simplify code and not get into NPE but probably it should get an NPE when you expect a value for entityId
I will drop this.